OpenRA.Mods.Common/Widgets/CommandBarButtonWidget.cs

```diff
@@ -0,0 +1,40 @@
+#region Copyright & License Information
+/*
+ * Copyright 2007-2017 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using System;
+using System.Drawing;
+using OpenRA.Widgets;
+
+namespace OpenRA.Mods.Common.Widgets
+{
+	public class CommandBarButtonWidget : ButtonWidget
+	{
+		[Translate] public string TooltipDesc;
+		public Func<string> GetTooltipDesc;
+		public Func<string> GetKeyName;
+
+		[ObjectCreator.UseCtor]
+		public CommandBarButtonWidget(ModData modData)
+			: base(modData)
+		{
+			GetTooltipDesc = () => TooltipDesc;
+			GetKeyName = () => null;
+		}
```

Comment:
You could add: OnKeyPress = ki => { if (ki.Event == KeyInputEvent.Down && !ki.IsRepeat) OnClick();
You can compare/hear the difference by holding A in release and this PR.

my suggestion from another PR: change default ButtonWidget.OnKeyPress from { OnClick(); } to {ki => { if (ki.Event == KeyInputEvent.Down && !ki.IsRepeat) OnClick(); }